src/cacheHttpClient.ts

```diff
@@ -144,7 +144,33 @@ export async function downloadCache(
     const stream = fs.createWriteStream(archivePath);
     const httpClient = new HttpClient("actions/cache");
     const downloadResponse = await httpClient.get(archiveLocation);
+
+    // Abort download if no traffic received over the socket.
+    downloadResponse.message.socket.setTimeout(SocketTimeout, () => {
+        downloadResponse.message.destroy();
+        core.debug(
+            `Aborting download, socket timed out after ${SocketTimeout} ms`
+        );
+    });
```

Comment:
What happens when this occurs? Is this when we would hit the mismatched file length below and throw the error?

Reply:
Yes, this closes the socket and interrupts the download, which is then caught by the file length check.